ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1104 @@
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Helper Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface MutableScopeWithWorkspace<TStorage> : IMutableScopeWithStorage<TStorage> ",
+                "\nwhere TStorage : ReportStorage",
+                "\n{",
+                "\n    protected IWorkspace workspace => GetStorage().Workspace;   ",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Currency Type"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface CurrencyFormsEntity : IMutableScope {",
+                "\n    [DropdownEnum(typeof(CurrencyType))]",
+                "\n    CurrencyType CurrencyType { get; set; } ",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Scenario"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface ScenarioFormsEntity<TStorage> : MutableScopeWithWorkspace<TStorage> ",
+                "\nwhere TStorage : ReportStorage ",
+                "\n{",
+                "\n    [DropdownMethod(nameof(GetScenarioAutocompleteAsync))]",
+                "\n    [Display(Name = \"Scenario\")]",
+                "\n    string ScenarioControl { get; set; }",
+                "\n",
+                "\n    [NotVisible] IDictionary<string, string> ScenarioMapping { get; set; }",
+                "\n    protected string Scenario => !string.IsNullOrWhiteSpace(ScenarioControl) && ",
+                "\n                                  ScenarioMapping is not null && ",
+                "\n                                  ScenarioMapping.TryGetValue(ScenarioControl, out var value) ",
+                "\n                                    ? value ",
+                "\n                                    : null;",
+                "\n    ",
+                "\n    async Task<IReadOnlyCollection<string>> GetScenarioAutocompleteAsync(string userInput, int page, int pageSize) {",
+                "\n        (ScenarioMapping, var orderedDropDownValues) = await workspace.GetAutocompleteMappings<Scenario>(true);",
```

Comment:
why is this not taken from the storage? it would be more efficient than querying each time. See also #237.